### DISCUSSION TITLE
How to create Go dynamic route

### DISCUSSION THREAD

**Discussion by veritem** [vercel/vercel#6469] [Help]
I'm working on a project, where i need to have a dynamic route like this `/api/book/1` is there anyway to do this with golang serverless on vercel thanks 🙏

**Comment by lalit2005** [comment]
I am really really new to golang. But if it were for nextjs, I could have told 😓 . I exactly don't know how to achieve this. But can you consider using query parameters instead of dynamic routes like so - `api/book/?id=1` instead of`api/book/1`?

**Reply by veritem** [reply]
Yeah that's what im doing currently, i'm wondering if there is a better way?

**Comment by barelyhuman** [comment]
@veritem 
You can use vercel.json => rewrites to declare definitions of the above to push to a certain go entry function

You can read more about it here.
https://vercel.com/docs/configuration#project/rewrites 

Overall 
1 .You setup the handler service to handle the params in normal go http syntax 
2. You add rewrites in vercel.json for the dynamic parameter. 
3. You respond based on the params. 

Hope that helps.
